numba/tests/test_function_type.py

```diff
@@ -593,9 +595,15 @@ def __init__(self, fname):
                 self._name = fname
                 if fname == 'cos':
                     addr = ctypes.cast(self.lib.cos, ctypes.c_voidp).value
-                    signature = float64(float64)
+                    if os.name == 'nt' and is32bits:
+                        signature = float32(float32)
+                    else:
+                        signature = float64(float64)
                 elif fname == 'sinf':
-                    addr = ctypes.cast(self.lib.sinf, ctypes.c_voidp).value
+                    if os.name == 'nt' and is32bits:
+                        addr = ctypes.cast(self.lib.sin, ctypes.c_voidp).value
+                    else:
+                        addr = ctypes.cast(self.lib.sinf, ctypes.c_voidp).value
```

Comment:
Might be good to add a couple of lines of comment about why all this is being done?

Reply:
Fixed.